modules/signing_worker/manifests/base.pp

```diff
@@ -0,0 +1,129 @@
+        path            => [ '/bin', '/usr/bin', '/usr/sbin', '/usr/local/bin', '/Library/Frameworks/Python.framework/Versions/3.7/bin'],
+    }
+
+
+    case $::hostname {
+        /^dep-mac-v3-signing\d+/: {
+            file { "${certs_dir}/widevine_dep.crt":
+                owner   => $signing_worker::user,
+                group   => $signing_worker::group,
+                content => base64('decode', lookup('signing_keys.widevine_dep_crt')),
+            }
+            file { "${certs_dir}/dep-signing.keychain":
+                owner   => $signing_worker::user,
+                group   => $signing_worker::group,
+                content => base64('decode', lookup('signing_keys.dep_signing_keychain')),
+            }
+        }
+        default: {
+            file { "${certs_dir}/widevine_prod.crt":
+                owner   => $signing_worker::user,
+                group   => $signing_worker::group,
+                content => base64('decode', lookup('signing_keys.widevine_prod_crt')),
+            }
+            file { "${certs_dir}/nightly_signing.keychain":
+                owner   => $signing_worker::user,
+                group   => $signing_worker::group,
+                content => base64('decode', lookup('signing_keys.nightly_signing_keychain')),
+            }
+            file { "${certs_dir}/release_signing.keychain":
+                owner   => $signing_worker::user,
+                group   => $signing_worker::group,
+                content => base64('decode', lookup('signing_keys.release_signing_keychain')),
+            }
+            file { "${certs_dir}/ed25519_privkey":
+                owner   => $signing_worker::user,
+                group   => $signing_worker::group,
+                content => lookup('signing_keys.ed25519_privkey'),
+                mode    => '0400',
+            }
+        }
+    }
+
+    $widevine_cert_path = $::hostname? {
+        /^dep-mac-v3-signing\d+/ => "${certs_dir}/widevine_dep.crt",
+        default => "${certs_dir}/widevine_prod.crt",
+    }
+    $ed_key_path = $::hostname? {
+        /^dep-mac-v3-signing\d+/ => '/dev/null',
+        default => "${certs_dir}/ed25519_privkey",
+    }
+
+    # scriptworker config
+    file { $script_config_file:
+        content => template('signing_worker/script_config.yaml.erb'),
+    }
+    file { $scriptworker_config_file:
+        content => template('signing_worker/scriptworker.yaml.erb'),
+    }
+
+    file { $scriptworker_wrapper:
+        content => template('signing_worker/scriptworker_wrapper.sh.erb'),
+        mode    => '0700',
+        owner   => $signing_worker::user,
+        group   => $signing_worker::group,
+    }
+
+    $launchd_script = "/Library/LaunchDaemons/org.mozilla.scriptworker.${signing_worker::user}.plist"
+    file { $launchd_script:
+        content => template('signing_worker/org.mozilla.scriptworker.plist.erb'),
+        mode    => '0644',
+    }
+    exec { "${signing_worker::user}_launchctl_load":
```

Comment:
We may want to be able to perform some checks before starting up. Should we have two levels of scripts (a check and a startup wrapper), or should we put those checks into the wrapper?

Reply:
What sort of checks? The startup wrapper will keep being run over and over until it stays running, so it depends how the checks should behave.

Reply:
e.g., don't start up if the widevine module isn't installed.